api/client.go

```diff
@@ -362,6 +397,17 @@ func (c *Client) Address() string {
 	return c.addr.String()
 }
 
+// SetLimiter will set the rate limiter for this client.
+// This method is thread-safe.
+// rateLimit and burst are specified according to https://godoc.org/golang.org/x/time/rate#NewLimiter
+func (c *Client) SetLimiter(rateLimit float64, burst int) {
+	c.modifyLock.RLock()
+	c.config.modifyLock.Lock()
+	defer c.modifyLock.RUnlock()
```

Comment:
Can we have the order of these two defers switched?

Reply:
^